frontend/src/pages/NewModel.tsx

```diff
@@ -1,96 +1,235 @@
+                                        type: 'locatorv1/hf',
+                                        repo: selectedModel,
+                                        file: selectedFile,
+                                    })
+                                        .unwrap()
+                                        .then(importJobId => {
+                                            setImportJob(importJobId);
+                                        })
+                                        .catch(error => {
+                                            // What to do about the error, update that we have a failed to import thing?
+                                            setImportJob(undefined);
+                                            setImportError(error);
+                                        });
+                                }
                             }}>
                             Register New Model
                         </button>
                     </div>
-                </form>
+                    {importJob &&
+                        (<div className="mt-5">
+                            <p className="text-gray-400/80">{importIsLoading && "loading..."} {importData ? JSON.stringify(importData) : ""}</p>
+                        </div>)
+                    }
+                </>
+            )}
+        </form>
+    )
+}
+
+export default function NewModel() {
+    const [selection, setSelection] = useState("none");
+
+    return (
+        <Page>
+            <OneColumnLayout type="narrow">
+                <div>
+                    <h3 className=" font-semibold text-3xl pb-1">Add a new model</h3>
+                    <p className=" text-lg text-white/70 pb-2">
+                        A model represents a set of binaries that are deployed to the server.
+                        Use this to register a new model with the server.
+                        Once deployed, models can be run from the Web Portal or via API.
+                    </p>
+                    <hr className=" border-primary-100" />
+                </div>
+
+                <div className="mt-4">
+                    <ButtonInput
+                        options={[
+                            {
+                                title: "Import from Disk",
+                                description: "Deploy a new model to the Web Server that is already available on your disk.",
+                                value: "disk"
+                            },
+                            {
+                                title: "Import from HuggingFace",
+                                description: "Select a model from HuggingFace to deploy to your Web Server.",
+                                value: "hugging-face"
+                            },
+                        ]}
+                        onSelect={setSelection}
+                    />
+                </div>
 
+                {selection == "none" && (<React.Fragment />)}
```

Comment:
nit: should use `===` when comparing strings

Reply:
for the other string comparisons below we should technically be using === as well but that sounds good to me!

Reply:
Addressed